internal/licenses/manager.go

```
@@ -0,0 +1,160 @@
+}
+
+func NewManager(store Store, config *Config, dataReader FileReaderFunc, verifier func(cert []byte) LicenseVerifier) Manager {
+	return &manager{
+		store:      store,
+		config:     config,
+		dataReader: dataReader,
+		verifier:   verifier,
+	}
+}
+
+func (m *manager) AddLicense(ctx context.Context, licenseFilePath string, licenseKey string, publicKey string) error {
+	slog.Debug("Starting to add a new license", "filePath", licenseFilePath)
+
+	cert, err := m.dataReader(licenseFilePath)
+	if err != nil {
+		slog.Error("failed to read license file", "filePath", licenseFilePath, "error", err)
+		return fmt.Errorf("failed to read license file: %w", err)
+	}
+
+	slog.Debug("Successfully read the license file", "filePath", licenseFilePath)
+
+	lic := m.verifier(cert)
+	keygen.PublicKey = publicKey
+
+	if err := lic.Verify(); err != nil {
+		slog.Error("license verification failed", "filePath", licenseFilePath, "error", err)
+		return fmt.Errorf("license verification failed: %w", err)
+	}
+
+	dec, err := lic.Decrypt(licenseKey)
+	if err != nil {
+		slog.Error("license decryption failed", "filePath", licenseFilePath, "error", err)
+		return fmt.Errorf("license decryption failed: %w", err)
+	}
+
+	id := dec.License.ID
+	key := dec.License.Key
+
+	if err := m.store.InsertLicense(ctx, id, cert, key); err != nil {
+		slog.Error("failed to insert license into store", "licenseID", id, "error", err)
+		return fmt.Errorf("failed to insert license: %w", err)
+	}
+
+	if m.config.EnabledAudit {
+		if err := m.store.InsertAuditLog(ctx, "add", "license", id); err != nil {
+			slog.Error("failed to insert audit log", "licenseID", id, "error", err)
+			return fmt.Errorf("failed to insert audit log: %w", err)
+		}
+	}
+
+	slog.Info("added license", "licenseID", id)
+	return nil
+}
+
+func (m *manager) RemoveLicense(ctx context.Context, id string) error {
+	slog.Debug("Starting to remove license", "id", id)
+
+	if err := m.store.DeleteLicenseByID(ctx, id); err != nil {
```

Can we return an error if the license doesn't exist?

---

Yes, now we will get an error if the record doesn`t exist. 
The removal executes [with transaction](https://github.com/keygen-sh/keygen-relay/pull/1/files#diff-96c04e9f3f5157f619cd787a5795242d9020c7b1a48d8cc21ff3fc4f9e62ad2dR36)